include/vigra/blockwise_labeling.hxx

```diff
@@ -108,21 +113,52 @@ blockwiseLabeling(DataBlocksIterator data_blocks_begin, DataBlocksIterator data_
         LabelBlocksIterator label_blocks_it = label_blocks_begin;
         typename MultiArray<Dimensions, Label>::iterator offsets_it = label_offsets.begin();
         Label current_offset = 0;
-        for( ; data_blocks_it != data_blocks_end; ++data_blocks_it, ++label_blocks_it, ++offsets_it)
-        {
-            vigra_assert(label_blocks_it != label_blocks_end && offsets_it != label_offsets.end(), "");
-            *offsets_it = current_offset;
-            if(background_value)
-            {
-                current_offset += 1 + labelMultiArrayWithBackground(*data_blocks_it, *label_blocks_it,
+        // a la OPENMP_PRAGMA FOR
+        
+        auto d = std::distance(data_blocks_begin, data_blocks_end);
+
+
+        std::vector<UInt32> nSeg(d);
+        //std::vector<int> ids(d);
+        //std::iota(ids.begin(), ids.end(), 0 );
+
+        parallel_foreach(nThreads,d,
+            [&](const int threadId, const uint64_t i){
+                int resVal;
+                if(background_value){
+                    resVal = 1 + labelMultiArrayWithBackground(data_blocks_it[i], label_blocks_it[i],
+                                                                        neighborhood, *background_value, equal);
+                }
+                else{
+                    resVal = labelMultiArray(data_blocks_it[i], label_blocks_it[i],
+                                                      neighborhood, equal);
+                }
```

Comment:
One of the assignments to `resVal` is probably wrong. Both `labelMultiArrayWithBackground()` and `labelMultiArray()` return the largest valid label. So, `1 +` should be used either in both cases or in neither case.